Sources/SwiftDocC/Model/Identifier.swift

```diff
@@ -489,6 +489,12 @@ public struct UnresolvedTopicReference: Hashable, CustomStringConvertible {
     /// An optional title.
     public var title: String? = nil
     
+    /// Optional candidates in case the reference couldn't be resolved.
+    ///
+    /// A resolution algorithm may attach possible candidates for an unresolvable reference to this field
+    /// for usage in diagnostics and fixits.
+    var canidates: [String] = []
```

Comment:
I replaced the `candidates` by a more sophisticated logic in `TopicReferenceResolutionError`. Essentially, `TopicReferenceResolutionError` now stores a list of `Solution`s that use relative locations. In order to extract these `Solution`s, one has to provide the location of the reference, so that the relative locations can be replaced by absolute ones.